tests/trainer/logging_/test_train_loop_logging.py

```diff
@@ -712,3 +712,35 @@ def training_step(self, *args):
     model = TestModel()
     with pytest.raises(MisconfigurationException, match=r'reduce_fx={min,max,mean,sum}\)` are currently supported'):
         trainer.fit(model)
+
+
+def test_sanity_metrics_are_reset_and_dont_appear_in_prog_bar(tmpdir):
```

Comment:
great! the test fails on master. However, I tried setting `num_sanity_val_steps=0` just to be sure but we get an error:

```
tmpdir = local('/private/var/folders/s2/338zrsts1b17v5cj18xr8_640000gn/T/pytest-of-adrian/pytest-108/test_sanity_metrics_are_reset_0')

    def test_sanity_metrics_are_reset_and_dont_appear_in_prog_bar(tmpdir):
    
        class TestModel(BoringModel):
    
            def validation_step(self, batch, batch_idx):
                output = super().validation_step(batch, batch_idx)
                if self.trainer.sanity_checking:
                    self.log("val_loss", output["x"], prog_bar=True, logger=True)
                return output
    
            def training_step(self, batch, batch_idx):
                loss = super().training_step(batch, batch_idx)
                if batch_idx == 0:
                    assert self.trainer.logger_connector._progress_bar_metrics == {}
                    assert self.trainer.logger_connector._logged_metrics == {}
                    assert self.trainer.logger_connector._callback_metrics == {}
                self.log("train_loss", loss, prog_bar=True, logger=True)
                return loss
    
        trainer = Trainer(
            default_root_dir=tmpdir,
            max_epochs=1,
            limit_train_batches=1,
            limit_val_batches=2,
            num_sanity_val_steps=0,
        )
>       trainer.fit(TestModel())

ddsaf.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../pytorch_lightning/trainer/trainer.py:522: in fit
    self._run(model)
../pytorch_lightning/trainer/trainer.py:883: in _run
    self._dispatch()
../pytorch_lightning/trainer/trainer.py:925: in _dispatch
    self.accelerator.start_training(self)
../pytorch_lightning/accelerators/accelerator.py:96: in start_training
    self.training_type_plugin.start_training(trainer)
../pytorch_lightning/plugins/training_type/training_type_plugin.py:161: in start_training
    self._results = trainer.run_stage()
../pytorch_lightning/trainer/trainer.py:935: in run_stage
    return self._run_train()
../pytorch_lightning/trainer/trainer.py:983: in _run_train
    self.fit_loop.run()
../pytorch_lightning/loops/base.py:91: in run
    self.advance(*args, **kwargs)
../pytorch_lightning/loops/fit_loop.py:207: in advance
    epoch_output = self.epoch_loop.run(train_dataloader)
../pytorch_lightning/loops/base.py:92: in run
    self.on_advance_end()
../pytorch_lightning/loops/epoch/training_epoch_loop.py:156: in on_advance_end
    self._run_validation()
../pytorch_lightning/loops/epoch/training_epoch_loop.py:181: in _run_validation
    self.trainer.fit_loop.val_loop.run()
../pytorch_lightning/loops/base.py:97: in run
    output = self.on_run_end()
../pytorch_lightning/loops/dataloader/evaluation_loop.py:147: in on_run_end
    self.on_evaluation_end()
../pytorch_lightning/loops/dataloader/evaluation_loop.py:212: in on_evaluation_end
    self.trainer.call_hook("on_validation_end", *args, **kwargs)
../pytorch_lightning/trainer/trainer.py:1147: in call_hook
    trainer_hook(*args, **kwargs)
../pytorch_lightning/trainer/callback_hook.py:227: in on_validation_end
    callback.on_validation_end(self, self.lightning_module)
../pytorch_lightning/callbacks/model_checkpoint.py:279: in on_validation_end
    self.save_checkpoint(trainer)
../pytorch_lightning/callbacks/model_checkpoint.py:323: in save_checkpoint
    monitor_candidates = self._monitor_candidates(trainer, epoch=epoch, step=global_step)
../pytorch_lightning/callbacks/model_checkpoint.py:678: in _monitor_candidates
    monitor_candidates = deepcopy(trainer.logger_connector.callback_metrics)
../../../.conda/envs/pl/lib/python3.8/copy.py:146: in deepcopy
    y = copier(x, memo)
../../../.conda/envs/pl/lib/python3.8/copy.py:230: in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
../../../.conda/envs/pl/lib/python3.8/copy.py:161: in deepcopy
    rv = reductor(4)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = {'loss': tensor(2.6577)}, drop_value = False

    def __getstate__(self, drop_value: bool = False) -> dict:
    
        def getstate(item: ResultMetric) -> dict:
            return item.__getstate__(drop_value=drop_value)
    
        items = apply_to_collection(dict(self), (ResultMetric, ResultMetricCollection), getstate)
>       return {"items": items, "meta": self.meta.__getstate__(), "_class": self.__class__.__name__}
E       AttributeError: 'NoneType' object has no attribute '__getstate__'

../pytorch_lightning/trainer/connectors/logger_connector/result.py:298: AttributeError
```

mind double check this?

Reply:
Hey @awaelchli,

Yes, got the same issue and resolved it with:

```
 if isinstance(value, ResultMetricCollection):
      value = dict(value.items())
```

We forgot to convert ResultMetricCollection to dictionaries.

Best,
T.C